### DISCUSSION TITLE
Is it possible to reduce the size after deploying kafka?

### DISCUSSION THREAD

**Discussion by shreyasarani23** [strimzi/strimzi-kafka-operator#4920] [General]
I have deployed kafka cluster with following configuration.
storage:
      type: persistent-claim
      size: 1Gi
      class: kafka-storage-class
      deleteClaim: true 
And I want to reduce the size to 500MB
storage:
      type: persistent-claim
      size: 500Mi
      class: kafka-storage-class
      deleteClaim: true 

Is it possible do it?

**Comment by scholzj** [comment]
Not easily. Kubernetes do not support shrinking of volumes. So you would need to do it manually at the infrastructure level. But keep in mind that Kafka is _storage hugry_. Even 1Gi storage is super small for Kafka unless you fine-tune all kind of parameters around log segments, retention etc.

**Reply by shreyasarani23** [reply]
Ok got it actually I am using it for dev/testing purposes.
According to you what are the most important/compulsory kafka configurations [including producer,conusmer,topic] that need to be specified in the yaml file so that our kafka cluster is efficient??

**Reply by scholzj** [reply]
Well, even with testing or development, 1 GB is really small ... you create few topics and it might be full.

You can check this PR https://github.com/strimzi/strimzi-kafka-operator/pull/4827 for some tuning tips. But there is no one-size-fits-all thing. So you need to figure out your self what exactly do you need.
